crates/bevy_reflect/src/path/access.rs

```diff
@@ -1,56 +1,74 @@
+//! Representation for individual element accesses within a path.
+
 use std::{borrow::Cow, fmt};
 
-use super::{AccessError, ReflectPathError};
+use super::ReflectPathError;
 use crate::{Reflect, ReflectMut, ReflectRef, VariantType};
 use thiserror::Error;
 
-type InnerResult<T> = Result<Option<T>, Error<'static>>;
+type InnerResult<'a, T> = Result<Option<T>, Error<'a>>;
 
+/// An error originating from an [`Access`] of an element within a type.
 #[derive(Debug, PartialEq, Eq, Error)]
-pub(super) enum Error<'a> {
+pub enum Error<'a> {
```

Comment:
Style: Bevy would call this `AccessError`. Some crates use the module name in the exports, but Bevy as a rule doesn't.